synapse/storage/database.py

```diff
@@ -1680,8 +1680,8 @@ async def simple_select_many_batch(
         self,
         table: str,
         column: str,
-        iterable: Iterable[Any],
-        retcols: Iterable[str],
+        iterable: Collection[Any],
```

Comment:
`iterable` should be a `Collection` since we check if it is truth-y, generators are always truth-y so by requiring a `Collection` the short circuit works better.

Reply:
this makes the comment at line 1710 a bit misleading.

Maybe we should just get rid of the short-circuit (it's not like the rest of the method is exactly heavy-weight for an empty iterable).

(Also, we could use `batch_iter` instead of rolling our own, less-efficient, version; `batch_iter` *does* support iterables.)

Reply:
I've made this change to use `batch_iter`.